exist-core/src/main/java/org/exist/dom/memtree/DocumentImpl.java

```diff
@@ -302,6 +302,13 @@ public void replaceReferenceNode(final int nodeNum, final CharSequence ch) {
         addChars(nodeNum, ch);
     }
 
+    /**
+     * @return TRUE when the Document is constructed via document {}.
+     */
+    public boolean isExplicitlyCreated(){
+        return explicitlyCreated;
```

Comment:
Hmmm.... Should this also determine the type returned from `#getNodeType()`. It would then be cheaper to call `Node#getNodeType()` than `node instance of`

Reply:
pooh..... lets discuss with @wolfgangmm ; I am open for improvements :-)

Reply:
@adamretter the document check is done Max 1 per call, so it will not be too expensive.
I tried to improve but it is (AFAICsee) impossible. And I do  now want spend a 4th day on this simple function :-)

When #1463 is implemented, we can reconsider.

For me this PR is good to go.

Reply:
checking S.O.: https://stackoverflow.com/questions/103564/the-performance-impact-of-using-instanceof-in-java 
performance is not an issue anymore in  modern JDKs.